src/workspaces/common/state/useWorkspaceStatePolling.ts

```diff
@@ -0,0 +1,118 @@
+import { LoadedState } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import { useEffect, useState } from 'react';
+import { Ajax } from 'src/libs/ajax';
+import { workspacesStore, workspaceStore } from 'src/libs/state';
+import { pollWithCancellation } from 'src/libs/utils';
+import { WorkspaceState, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+const updateWorkspacesList = (
```

Comment:
Is it possible to write unit tests for this code?

Reply:
There are already tests that hit this functionality directly, in `WorkspaceList`.

It's just easier to test in the larger component, rather than creating some test component for the hook.

Reply:
There must be sections that aren't covered though by the existing tests: https://sonarcloud.io/component_measures?id=DataBiosphere_terra-ui&pullRequest=4819&metric=new_coverage&view=list

I don't care too much about whether you test it through `WorkspaceList` or in its own test, but you might find it easier to write a targeted test for the hook function.